core/data/src/main/java/uk/gov/gchq/gaffer/data/element/function/ElementAggregator.java

```diff
@@ -57,6 +60,19 @@ public Properties apply(final Properties state, final Properties properties) {
     }
 
     @Override
+    public List<TupleAdaptedBinaryOperator<String, ?>> getComponents() {
+        if (readOnly) {
+            return Collections.unmodifiableList(super.getComponents());
+        }
+
+        return super.getComponents();
+    }
+
+    public void lock() {
```

Comment:
No, I think once the ElementAggregator is locked - no changes should be allowed. If you need to make changes then you should create a new instance of ElementAggregator.